### PR TITLE
Replace glide novendor, fix gosec lint to exclude crossdock

### LINKED ISSUE #1097: gosec fails on crossdock
## Requirement - what kind of business use case are you trying to solve?

Cannot submit new pull requests to GitHub.

## Problem - what in Jaeger blocks you from solving the requirement?

gosec fails on crossdock source code. Causes Travis CI errors (https://travis-ci.org/jaegertracing/jaeger/jobs/434740163).

Example:

```
[/Users/isaachier/gocode/src/github.com/jaegertracing/jaeger/crossdock/main.go:92] - G107: Potential HTTP request made with variable url (Confidence: MEDIUM, Severity: MEDIUM)
  > http.Get(healthURL)
```

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Instead of excluding error G107, which should be fine in normal Jaeger source code, we should exclude the crossdock directory from unnecessary linting.

### PR DESCRIPTION
**Pull request by isaachier**
- Resolves #1097.

- Exclude crossdock from gosec
- Replace `glide novendor` with custom bash script

### CODE REVIEW

**Comment by isaachier** [comment]
Test needs to be rerun.

File: scripts/list-packages.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+		-not -path './thrift-gen' \
+		-not -path './swagger-gen' \
+		-not -path './examples' \
+		-not -path './scripts')
+	for d in $dirs; do
+		find "$d" -name '*.go' -type f -exec echo "$d/..." \; -quit
+	done
+}
+
+listPackages "$1"
```
Comment (black-adder) [inline scripts/list-packages.sh]: is this variable being used in the function?

File: scripts/list-packages.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	local d
+	local dirs
+	dirs=$(find . -mindepth 1 -maxdepth 1 -type d \
+		-not -path './vendor' \
+		-not -path './thrift-gen' \
+		-not -path './swagger-gen' \
+		-not -path './examples' \
+		-not -path './scripts')
+	for d in $dirs; do
+		find "$d" -name '*.go' -type f -exec echo "$d/..." \; -quit
```
Comment (black-adder) [inline scripts/list-packages.sh]: why is the `/...` needed after the filename?

Comment (isaachier) [reply scripts/list-packages.sh]: It's what glide does. More importantly, I believe it includes all subpackages.

File: scripts/list-packages.sh (hunk 1 above)
Comment (isaachier) [reply scripts/list-packages.sh]: My bad. Should have used this instead of current directory in when invoking `find`.

**Pull request closed without merging**